distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Period.enso

```diff
@@ -0,0 +1,146 @@
+from Standard.Base import all
+
+import Standard.Base.Data.Time.Duration
+
+polyglot java import java.time.Period as Java_Period
+polyglot java import java.time.DateTimeException
+polyglot java import java.lang.ArithmeticException
+
+
+## Create a Period representing the time interval between two dates.
+
+   Arguments:
+   - start_date_inclusive: The start date of the period, included.
+   - end_date_exclusive: The end date of the period, excluded.
+
+   > Example
+     Get a Period between 2022-10-21 and 2022-09-12
+
+         import Standard.Base.Data.Time.Period
+
+         example_period = Period.between (Date.new 2022 10 21) (Date.new 2022 9 12)
+between : Date -> Date -> Period
+between start_date_inclusive end_date_exclusive =
+    Period.Period_Data (Java_Period.between start_date_inclusive end_date_exclusive)
+
+## Create a new Period from years, months and days.
+
+   Arguments:
+   - years: Amount of years.
+   - months: Amount of months.
+   - days: Amount of days.
+
+   > Example
+     Create a Period of 2 years and 5 days
+
+         import Standard.Base.Data.Time.Period
+
+         example_period = Period.new 2 0 5
+new : Integer -> Integer -> Integer -> Period
+new years=0 months=0 days=0 =
+    Period.Period_Data (Java_Period.of years months days)
+
+## PRIVATE
+ensure_period : Any -> Text -> Period ! (Time_Error | Illegal_Argument_Error)
+ensure_period object error_msg="Cannot use Duration as a parameter" =
+    case object of
+        period : Period -> period
+        _ : Duration.Duration ->
+            Error.throw Time_Error_Data error_msg
+        x : Any ->
```

Comment:
```suggestion
        x ->
```

minor nitpick: I think we usually just avoid the type ascription altogether for the catchall branch.